pennylane/gradients/hadamard_gradient.py

```diff
@@ -114,72 +104,94 @@ def hadamard_grad(
-     tensor([-0.38355704], requires_grad=True))
-
-    This quantum gradient transform can also be applied to low-level
-    :class:`~.QuantumTape` objects. This will result in no implicit quantum
-    device evaluation. Instead, the processed tapes, and post-processing
-    function, which together define the gradient are directly returned:
-
-    >>> ops = [qml.RX(p, wires=0) for p in params]
-    >>> measurements = [qml.expval(qml.PauliZ(0))]
-    >>> tape = qml.tape.QuantumTape(ops, measurements)
-    >>> gradient_tapes, fn = qml.gradients.hadamard_grad(tape)
-    >>> gradient_tapes
-    [<QuantumTape: wires=[0, 1], params=3>,
-     <QuantumTape: wires=[0, 1], params=3>,
-     <QuantumTape: wires=[0, 1], params=3>]
-
-    This can be useful if the underlying circuits representing the gradient
-    computation need to be analyzed.
-
-    The output tapes can then be evaluated and post-processed to retrieve
-    the gradient:
-
-    >>> dev = qml.device("default.qubit", wires=2)
-    >>> fn(qml.execute(gradient_tapes, dev, None))
-    (array(-0.3875172), array(-0.18884787), array(-0.38355704))
-
     This transform can be registered directly as the quantum gradient transform
     to use during autodifferentiation:
 
-    >>> dev = qml.device("default.qubit", wires=3)
+    >>> import jax
+    >>> dev = qml.device("default.qubit", wires=2)
     >>> @qml.qnode(dev, interface="jax", diff_method="hadamard")
     ... def circuit(params):
     ...     qml.RX(params[0], wires=0)
     ...     qml.RY(params[1], wires=0)
     ...     qml.RX(params[2], wires=0)
-    ...     return qml.expval(qml.PauliZ(0))
-    >>> params = jax.numpy.array([0.1, 0.2, 0.3])
-    >>> jax.jacobian(circuit)(params)
-    [-0.3875172  -0.18884787 -0.38355704]
-
-    If you use custom wires on your device, you need to pass an auxiliary wire.
-
-    >>> dev_wires = ("a", "c")
-    >>> dev = qml.device("default.qubit", wires=dev_wires)
-    >>> @qml.qnode(dev, interface="jax", diff_method="hadamard", aux_wire="c", device_wires=dev_wires)
-    >>> def circuit(params):
-    ...    qml.RX(params[0], wires="a")
-    ...    qml.RY(params[1], wires="a")
-    ...    qml.RX(params[2], wires="a")
-    ...    return qml.expval(qml.PauliZ("a"))
+    ...     return qml.expval(qml.PauliZ(0)), qml.var(qml.PauliZ(0))
     >>> params = jax.numpy.array([0.1, 0.2, 0.3])
     >>> jax.jacobian(circuit)(params)
-    [-0.3875172  -0.18884787 -0.38355704]
+    (Array([-0.38751727, -0.18884793, -0.3835571 ], dtype=float32),
+    Array([0.6991687 , 0.34072432, 0.6920237 ], dtype=float32))
+
+    .. details::
+        :title: Usage Details
+
+        This gradient transform can be applied directly to :class:`QNode <pennylane.QNode>` objects. This is not
+        recommended because PennyLane must compute the classical Jacobian of the parameters and multiply it with
+        the quantum Jacobian, we recommend using the ``diff_method`` kwargs with your favorite machine learning
+        framework.
```

Comment:
Minor comment @rmoyard, but I would simplify this statement here (and elsewhere where it appears):

```rest
This gradient transform can be applied directly to :class:`QNode <pennylane.QNode>` objects. However, in general we recommend providing
the gradient transform as the ``diff_method`` argument of the QNode
decorator, and differentiating with your preferred machine learning
framework.
```

I think there is not a need to talk about the implementation details (classical/quantum jacobians) here.

Reply:
```
        This gradient transform can be applied directly to :class:`QNode <pennylane.QNode>` objects.
        However, for performance reasons, we recommend providing the gradient transform as the ``diff_method`` argument 
        of the QNode decorator, and differentiating with your preferred machine learning framework.
```